ntfysh_client/NotificationDialog.cs

```diff
@@ -0,0 +1,263 @@
+        {
+            base.OnShown(e);
+        }
+
+        private void SetWindowPosition()
+        {
+            int workingtop = Screen.PrimaryScreen.WorkingArea.Height - this.Height;
+            this.Top = workingtop - NotificationDialog.ScreenMargin;
+
+            int workingleft = Screen.PrimaryScreen.WorkingArea.Width - this.Width;
+            this.Left = workingleft - NotificationDialog.ScreenMargin;
+        }
+
+        private void ui_hide_window(object? sender, EventArgs? e)
+        {
+            AnimateWindow(
+                this.Handle,
+                time: 250,
+                flags: NFWinUserAnimateWindowConstnats.AW_SLIDE | NFWinUserAnimateWindowConstnats.AW_VER_POSITIVE | NFWinUserAnimateWindowConstnats.AW_HIDE
+            );
+            this.IsVisible = false;
+        }
+
+        private void handleTimeout(object? sender, EventArgs? e)
+        {
+            this.cancelTimer();
+            if (this.InvokeRequired)
+            {
+                // on a background thread, so invoke on the UI thread
+                this.Invoke(new Action(() => this.ui_hide_window(sender, e)));
+            }
+            else
+            {
+                // in the UI thread, invoke directly
+                this.ui_hide_window(sender, e);
+            }
+        }
+
+        private Image? ConvertToolTipIconToImage(ToolTipIcon icon)
+        {
+            switch (icon)
+            {
+                case ToolTipIcon.Info:
+                    return SystemIcons.Information.ToBitmap();
+                case ToolTipIcon.Warning:
+                    return SystemIcons.Warning.ToBitmap();
+                case ToolTipIcon.Error:
+                    return SystemIcons.Error.ToBitmap();
+                case ToolTipIcon.None:
+                default:
+                    return null;
+            }
+        }
+
+        private void InitializeWindowHidden()
+        {
+            this.Opacity = 0;
+            this.ShowNotification("Title", "Message");
+            this.IsVisible = false;
+            this.Opacity = 1;
+        }
+
+        private void btnClose_Click(object sender, EventArgs e)
+        {
+            // immediate
+            this.IsVisible = false;
+        }
+
+        private class NFWinUserAnimateWindowConstnats
+        {
+            public const int AW_HOR_POSITIVE = 0x00000001;
+            public const int AW_HOR_NEGATIVE = 0x00000002;
+            public const int AW_VER_POSITIVE = 0x00000004;
+            public const int AW_VER_NEGATIVE = 0x00000008;
+            public const int AW_CENTER = 0x00000010;
+            public const int AW_HIDE = 0x00010000;
+            public const int AW_ACTIVATE = 0x00020000;
+            public const int AW_SLIDE = 0x00040000;
+            public const int AW_BLEND = 0x00080000;
+        }
+
+        private void window_MouseDown(object sender, EventArgs e)
+        {
+            this.cancelTimer();
+        }
+
+        private void cancelTimer()
```

Comment:
Nitpick: Naming convention is capitalised first letter

Reply:
renamed